source/common/ssl/context_config_impl.h

```diff
@@ -60,6 +71,12 @@ class ContextConfigImpl : public virtual Ssl::ContextConfig {
       cvc_update_callback_handle_ =
           certficate_validation_context_provider_->addUpdateCallback(callback);
     }
+    if (trusted_ca_provider_) {
+      if (tca_update_callback_handle_) {
+        tca_update_callback_handle_->remove();
+      }
+      tca_update_callback_handle_ = trusted_ca_provider_->addUpdateCallback(callback);
+    }
```

Comment:
We have 3 handlers,  it is time to use a list for them.  Maybe define a struct to hold {provider_,  and handler_} and use a list to store them